### PR TITLE
Update foreman package dependencies

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

File: packages/foreman/foreman/foreman.spec [hunk 1]
```diff
@@ -158,13 +159,15 @@ BuildRequires: %{?scl_prefix_ruby}rubygem(rake) >= 0.8.3
 BuildRequires: %{?scl_prefix_ruby}rubygem(rdoc)
 BuildRequires: %{?scl_prefix}rubygem(bundler_ext)
 # start specfile sqlite BuildRequires
```
Comment (ekohl) [inline packages/foreman/foreman/foreman.spec:161]: Perhaps we need a fix to start dropping these?

Comment (evgeni) [reply packages/foreman/foreman/foreman.spec:161]: Agreed. But that should happen in core first?

Comment (tbrisker) [reply packages/foreman/foreman/foreman.spec:161]: IIRC we decided to first make sure everything works properly with nulldb, but looks like it does, so definitely.

File: packages/foreman/foreman/foreman.spec [hunk 2]
```diff
@@ -158,13 +159,15 @@ BuildRequires: %{?scl_prefix_ruby}rubygem(rake) >= 0.8.3
...
-BuildRequires: %{?scl_prefix}rubygem(sqlite3) < 1.4.0
+BuildRequires: %{?scl_prefix}rubygem(sqlite3) >= 1.4
+BuildRequires: %{?scl_prefix}rubygem(sqlite3) < 2.0
 # end specfile sqlite BuildRequires
 
 
 # start specfile main BuildRequires
-BuildRequires: %{?scl_prefix}rubygem(rails) = 5.2.1
+BuildRequires: %{?scl_prefix}rubygem(cgi) >= 0.a
+BuildRequires: %{?scl_prefix}rubygem(yaml) >= 0.a
```
Comment (tbrisker) [inline packages/foreman/foreman/foreman.spec:169]: Where did these come from?

File: packages/foreman/foreman/foreman.spec (hunk 1 above)
Comment (ekohl) [reply packages/foreman/foreman/foreman.spec:161]: I was indeed saying that it should happen in core and then packaging will follow.

File: packages/foreman/foreman/foreman.spec (hunk 2 above)
Comment (ekohl) [reply packages/foreman/foreman/foreman.spec:169]: The script uses rubygems and internally that also calls the `gem()` method. Perhaps Ruby loads these internally

**Comment by ekohl** [comment]
```
DEBUG util.py:598:  Error: No Package found for tfm-rubygem(cgi) >= 0.a
DEBUG util.py:598:  Error: No Package found for tfm-rubygem(coffee-rails) >= 5.0.0
DEBUG util.py:598:  Error: No Package found for tfm-rubygem(rails-i18n) >= 6.0
DEBUG util.py:598:  Error: No Package found for tfm-rubygem(sqlite3) >= 1.4
DEBUG util.py:598:  Error: No Package found for tfm-rubygem(yaml) >= 0.a
```

I assume the `0.a` versions are something Ruby internal and I should drop them. The other 3 appear to be real issues that need to be updated.

File: packages/foreman/foreman/foreman.spec (hunk 1 above)
Comment (tbrisker) [reply packages/foreman/foreman/foreman.spec:161]: opened https://github.com/theforeman/foreman/pull/7617 for core

**Comment by tbrisker** [comment]
@zjhuntin looks like the gems that need newer versions are `coffee-rails` and `rails-i18n`, we're dropping `sqlite`.

**Comment by ekohl** [comment]
Blocked on https://github.com/theforeman/foreman-packaging/pull/5123

**Comment by zjhuntin** [comment]
[test rpm]

**Comment by tbrisker** [comment]
Still needs new `coffee-rails` and `rails-i18n` builds:
```
DEBUG util.py:598:  Error: No Package found for tfm-rubygem(coffee-rails) >= 5.0.0
DEBUG util.py:598:  Error: No Package found for tfm-rubygem(rails-i18n) >= 6.0
DEBUG util.py:598:  Error: No Package found for tfm-rubygem(sqlite3) >= 1.4
```
I think you can go ahead and drop sqlite, the pr dropping it in core should also be ready: https://github.com/theforeman/foreman/pull/7617

**Comment by tbrisker** [comment]
@ekohl can you please rebase this on top of https://github.com/theforeman/foreman-packaging/pull/5132 and https://github.com/theforeman/foreman-packaging/pull/4814 ?

**Comment by ekohl** [comment]
So to unblock this, we need:
* https://github.com/theforeman/foreman/pull/7621 merged or 0e8bdd09c4ad5cb615a2ed8672073ef38f96303a reverted
* https://github.com/theforeman/foreman/pull/7187 merged or b0018b43e451fc1a9c5afde2b7ffd612874db894 reverted

Once we've fixed the above 2 (my preference is to merge core), I'll update this PR.

**Comment by tbrisker** [comment]
I would prefer merging patternfly and not sprockets, as that one requires more changes than just dependency, if that is possible.

**Comment by mmoll** [comment]
patternfly got merged, as agreed.

**Comment by ekohl** [comment]
So I cheated a bit and used the old sqlite lower version bound so we can drop it later. The babel pins do break things since we don't explicitly package those but rather depend on `theforeman-builder`. Oh how I hate NPM with so much passion.

**Comment by ekohl** [comment]
Local changes compared to foreman.git:
* Bumped rails from 6.0.2.1 to 6.0.2.2
* Dropped `@babel/preset-env` and other exact pins that we pull in via `theforeman/builder`
* Lowered the sqlite version dependency since https://github.com/theforeman/foreman-packaging/pull/5132 will drop it and we use nulldb anyway.

**Comment by ekohl** [comment]
[test rpm] because koji was still building dependencies when I pushed.

**Comment by ekohl** [comment]
```
DEBUG util.py:598:  Error: Package: tfm-rubygem-actionpack-6.0.2.2-1.el7.noarch (build)
DEBUG util.py:598:             Requires: rh-ruby25-rubygem(rack) >= 2.0.8
DEBUG util.py:598:  Error: Package: tfm-rubygem-actionpack-6.0.2.2-1.el7.noarch (build)
DEBUG util.py:598:             Requires: rh-ruby25-rubygem(rack) < 3
DEBUG util.py:598:  Error: Package: tfm-rubygem-actionpack-6.0.2.2-1.el7.noarch (build)
DEBUG util.py:598:             Requires: rh-ruby25-rubygem(rack) >= 2.0
```

Maybe SCL doesn't provide rack anymore and we do have a package, so this needs an update:
https://github.com/theforeman/foreman-packaging/blob/b35819b1c6d790d8fdc23da762dd7317b0bda492/scl_prefixes.json#L5

```
DEBUG util.py:598:  Error: Package: tfm-rubygem-actioncable-6.0.2.2-1.el7.noarch (build)
DEBUG util.py:598:             Requires: tfm-rubygem(actionpack) = 5.2.1
DEBUG util.py:598:             Installing: tfm-rubygem-actionpack-6.0.2.2-1.el7.noarch (build)
DEBUG util.py:598:                 tfm-rubygem(actionpack) = 6.0.2.2
```

Bad update in 1431d2d6e3111a09f9bb2b7b2136b4687c11cdd2; https://rubygems.org/gems/actioncable/versions/6.0.2.2 states it needs 6.0.2.2. @zjhuntin did you use `obal update` instead of `bump_rpm.sh` again?

```
DEBUG util.py:598:  Error: Package: tfm-rubygem-roadie-rails-2.0.0-3.el7.noarch (build)
DEBUG util.py:598:             Requires: tfm-rubygem(railties) < 5.3
DEBUG util.py:598:             Installing: tfm-rubygem-railties-6.0.2.2-1.el7.noarch (build)
DEBUG util.py:598:                 tfm-rubygem(railties) = 6.0.2.2
```

Same story as actioncable.

```
DEBUG util.py:598:  Error: Package: tfm-rubygem-nokogiri-1.8.4-8.el7.x86_64 (build)
DEBUG util.py:598:             Requires: tfm-rubygem(mini_portile2) < 2.4
DEBUG util.py:598:             Available: tfm-rubygem-mini_portile2-2.4.0-1.el7.noarch (build)
DEBUG util.py:598:                 tfm-rubygem(mini_portile2) = 2.4.0
DEBUG util.py:598:  Error: Package: tfm-rubygem-nokogiri-1.8.4-8.el7.x86_64 (build)
DEBUG util.py:598:             Requires: tfm-rubygem(mini_portile2) < 2.4
DEBUG util.py:598:             Installing: tfm-rubygem-mini_portile2-2.4.0-1.el7.noarch (build)
DEBUG util.py:598:                 tfm-rubygem(mini_portile2) = 2.4.0
DEBUG util.py:598:  Error: Package: tfm-rubygem-rails-i18n-6.0.0-1.el7.noarch (build)
DEBUG util.py:598:             Requires: tfm-rubygem(railties) < 6.0
DEBUG util.py:598:             Installing: tfm-rubygem-railties-6.0.2.2-1.el7.noarch (build)
DEBUG util.py:598:                 tfm-rubygem(railties) = 6.0.2.2
DEBUG util.py:598:  Error: Package: tfm-rubygem-actiontext-6.0.2.2-1.el7.noarch (build)
DEBUG util.py:598:             Requires: tfm-rubygem(nokogiri) >= 1.8.5
DEBUG util.py:598:             Installing: tfm-rubygem-nokogiri-1.8.4-8.el7.x86_64 (build)
DEBUG util.py:598:                 tfm-rubygem(nokogiri) = 1.8.4
DEBUG util.py:598:  Error: Package: tfm-rubygem-actiontext-6.0.2.2-1.el7.noarch (build)
DEBUG util.py:598:             Requires: tfm-rubygem(nokogiri) >= 1.8.5
DEBUG util.py:598:             Available: tfm-rubygem-nokogiri-1.8.4-8.el7.x86_64 (build)
DEBUG util.py:598:                 tfm-rubygem(nokogiri) = 1.8.4
```

Nokogiri just looks busted. 7cc63e3ada29501ce3005f4426605efe3b32e1dc removed the gem file that was used in .spec. eb6690230abb3ca5bae80d67a2936b20c2a5601d did add 1.10.7, but didn't bump the version itself. How did this pass CI?

**Comment by zjhuntin** [comment]
I did not, I only used the bump_rpm script, Nokogiri hasn't been merged yet becuase it's not totally ready: https://github.com/theforeman/foreman-packaging/pull/5090, I'm not sure why actioncable is that way, i'll retake a look at it.

**Comment by ekohl** [comment]
Just to keep it central:
* b3c154a268dd0cc1d7e247224ebadbb706fc2504 fixed nokogiri
* e3cec0601eefabe5dd73980872e021a42bf5aa04 fixed actioncable

Once rack and road-rails are fixed as well, this can be retested.

**Comment by ekohl** [comment]
https://github.com/theforeman/foreman-packaging/pull/5142 checks roadie-rails off the list.

**Comment by tbrisker** [comment]
[test rpm]

**Comment by tbrisker** [comment]
@zjhuntin looks like you have a few more gems to update requirements for:
```
DEBUG util.py:598:  Error: Package: tfm-rubygem-actionpack-6.0.2.2-1.el7.noarch (build)
DEBUG util.py:598:             Requires: rh-ruby25-rubygem(rack) >= 2.0
DEBUG util.py:598:  Error: Package: tfm-rubygem-rails-i18n-6.0.0-1.el7.noarch (build)
DEBUG util.py:598:             Requires: tfm-rubygem(railties) < 6.0
DEBUG util.py:598:             Installing: tfm-rubygem-railties-6.0.2.2-1.el7.noarch (build)
DEBUG util.py:598:                 tfm-rubygem(railties) = 6.0.2.2
DEBUG util.py:598:  Error: Package: tfm-rubygem-actionpack-6.0.2.2-1.el7.noarch (build)
DEBUG util.py:598:             Requires: rh-ruby25-rubygem(rack) >= 2.0.8
DEBUG util.py:598:  Error: Package: tfm-rubygem-roadie-rails-2.1.1-1.el7.noarch (build)
DEBUG util.py:598:             Requires: tfm-rubygem(railties) < 5.3
DEBUG util.py:598:             Installing: tfm-rubygem-railties-6.0.2.2-1.el7.noarch (build)
DEBUG util.py:598:                 tfm-rubygem(railties) = 6.0.2.2
DEBUG util.py:598:  Error: Package: tfm-rubygem-actionpack-6.0.2.2-1.el7.noarch (build)
DEBUG util.py:598:             Requires: rh-ruby25-rubygem(rack) < 3

**Comment by tbrisker** [comment]
[test rpm]

**Comment by ekohl** [comment]
Rebased to clear out all the failed builds. Looks like it can now actually get to the build process so that's good.

**Comment by ekohl** [comment]
Both EL7 and EL8 fail:
```
NameError: uninitialized constant ActionView::Template::Handlers::Rabl
/builddir/build/BUILD/foreman-2.1.0-develop/config/initializers/rabl_init.rb:67:in `<top (required)>'
/opt/theforeman/tfm/root/usr/share/gems/gems/railties-6.0.2.2/lib/rails/engine.rb:667:in `block in load_config_initializer'
```

**Comment by mmoll** [comment]
Looks better now, AFAICT.

**Comment by ehelms** [comment]
Repoclosure LGTM -- nightly will just need https://github.com/theforeman/foreman-packaging/pull/5141 as well to continue past repoclosure

**Pull request merged**